kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/service/impl/LoginServiceImpl.java

```diff
@@ -214,6 +215,44 @@ public AccessTokenResponseDTO loginRedirect(String state, String sessionState, S
 		return accessTokenResponseDTO;
 	}
 
+	private String getClientAssertion() {
+		JWSSignatureRequestDto jwsSignatureRequestDto = new JWSSignatureRequestDto();
+		try {
+			jwsSignatureRequestDto.setDataToSign(getDataToSign());
+			jwsSignatureRequestDto.setReferenceId(this.environment.getProperty(Constants.CLIENT_ASSERTION_REFERENCE_ID));
+			jwsSignatureRequestDto.setApplicationId(this.environment.getProperty(Constants.APPLICATION_ID));
+			jwsSignatureRequestDto.setIncludePayload(Boolean.valueOf(this.environment.getProperty(Constants.IS_INCLUDE_PAYLOAD)));
+			jwsSignatureRequestDto.setIncludeCertificate(Boolean.valueOf(this.environment.getProperty(Constants.IS_INCLUDE_CERTIFICATE)));
+			jwsSignatureRequestDto.setIncludeCertHash(Boolean.valueOf(this.environment.getProperty(Constants.IS_iNCLUDE_CERT_HASH)));
+
+			RequestWrapper<JWSSignatureRequestDto> requestWrapper = new RequestWrapper<>();
+			requestWrapper.setRequest(jwsSignatureRequestDto);
+			requestWrapper.setRequesttime(DateUtils.getUTCCurrentDateTime());
+			UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromUriString(Objects.requireNonNull(this.environment.getProperty(Constants.KEYMANAGER_JWT_SIGN_END_POINT)));
+			JWTSignatureResponseDto jwtSignatureResponseDto = restTemplate.postForEntity(uriBuilder.toUriString(), requestWrapper, JWTSignatureResponseDto.class).getBody();
+			return Objects.requireNonNull(jwtSignatureResponseDto).getJwtSignedData();
+		} catch (HttpClientErrorException | HttpServerErrorException e) {
+			throw new ServiceException(Errors.JWT_SIGN_EXCEPTION.getErrorCode(),
+					Errors.JWT_SIGN_EXCEPTION.getErrorMessage());
+		}
+	}
+
+	private String getDataToSign() {
+		Map dataToSignMap = new LinkedHashMap();
+		String mosipResidentIDPClient = this.environment.getProperty(Constants.MOSIP_RESIDENT_IDP_CLIENT_ID);
+		dataToSignMap.put(Constants.SUB, mosipResidentIDPClient);
+		dataToSignMap.put(Constants.ISS, mosipResidentIDPClient);
+		dataToSignMap.put(Constants.AUD, Constants.BASE_URL);
+		dataToSignMap.put(Constants.EXP, getExpiryTime());
+		dataToSignMap.put(Constants.IAT, DateUtils.getUTCCurrentDateTimeString());
+		return CryptoUtil.encodeToPlainBase64(dataToSignMap.toString().getBytes());
```

Comment:
Convert the map to JSON using ObjectMapper.

Reply:
done.